### ISSUE TITLE
End-of-Life for Node.js 16

### ISSUE THREAD

**Issue by jalondonot** [openedx/wg-build-test-release#254]
https://nodejs.org/en/blog/announcements/nodejs16-eol/

**Comment by ghassanmas** [comment]
There is also an issue opened for this openedx/platform-roadmap/issues/243 @jmbowman do you have any suggestions or input if FED-BOM is expected to handle this completely?

**Comment by jmbowman** [comment]
The plan is to follow the [Upgrade Project Runbook](https://openedx.atlassian.net/wiki/spaces/AC/pages/3660316693/Upgrade+Project+Runbook), which is still being revised based on incoming feedback.  So issues will be created for the work needed in each repo, and the maintainers of those repos will be consulted regarding how much of the upgrade work they would like to do themselves vs. delegating to FED-BOM.  And FED-BOM is encouraged to delegate work to the broader Open edX developer community where appropriate.

**Comment by jfavellar90** [comment]
@jalondonot It seems it requires a lot of work to migrate from Node16 before the Palm cutoff. This issue should be a high priority, what do you think? Should we open a discussion in the OpenedX discuss?

**Comment by arbrandes** [comment]
@jfavellar90, we have it on our radar, yes.  It's [on the roadmap](https://github.com/openedx/platform-roadmap/issues/243), as a matter of fact.  As you can see there, FED-BOM is tasked with handling it.  The intention is to finish it before the Palm cut-off, but they might require help.  I asked them to holler if that happens.

In any case, it's worth keeping an eye on that roadmap ticket as the cut-off date nears.

**Comment by ghassanmas** [comment]
The issue/ticjet though doesn't mention edx-platform, was that intentional?, is it in the scope of the ticket?

**Comment by arbrandes** [comment]
@ghassanmas, good question.  Do you mind adding that question to the roadmap ticket?

**Comment by mariajgrimaldi** [comment]
Hi there @arbrandes! Do we know how is this issue going? Let us know!

**Comment by jmbowman** [comment]
The status in the tasklist of the roadmap issue is accurate: https://github.com/openedx/platform-roadmap/issues/243 .  Most of the Node-using repos were already updated to use Node 18 in time for the Palm release.  The only stragglers are credentials (in the process of migrating to a new UI and throwing the old one away), edx-platform (blocked by other in-progress maintenance and upgrades), and edx-analytics-dashboard (CI failing with the changes, debugging is in progress).  The security impact is relatively minor as these only use Node as a build-time dependency during deployments, and we plan to have more (ideally all) of the stragglers completed in time for the Quince release.

**Comment by arbrandes** [comment]
This has now been superseded by https://github.com/openedx/public-engineering/issues/229.
